extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/JaegerPropagator.java

```diff
@@ -115,14 +119,20 @@ public static JaegerPropagator getInstance() {
     setter.set(carrier, PROPAGATION_HEADER, new String(chars));
   }
 
-  private static <C> void injectBaggage(Baggage baggage, C carrier, TextMapSetter<C> setter) {
+  private static <C> void injectBaggage(
+      Baggage baggage, @Nullable C carrier, TextMapSetter<C> setter) {
     baggage.forEach(
         (key, baggageEntry) -> setter.set(carrier, BAGGAGE_PREFIX + key, baggageEntry.getValue()));
   }
 
   @Override
   public <C> Context extract(Context context, @Nullable C carrier, TextMapGetter<C> getter) {
-    Objects.requireNonNull(getter, "getter");
+    if (context == null) {
+      return Context.root();
```

Comment:
should this return `null`? Not sure what is the right behavior here.

Reply:
not sure either, but not returning null seems like a good thing in general. And, I can definitely understand "you passed in no Context, you get an empty one by default" as a reasonable fallback.

Reply:
+1 on not returning null

Reply:
But you passed null... it is not like we return null from nowhere

Reply:
methods that never return null are very nice, because it's one less thing callers need to worry about

if you do want to return null, even in a small case like this, the method return type should be annotated with `@Nullable`

Reply:
As with all our safety checks, the behavior is undefined so we can pick anything really. I'd lean towards root context since it seems to have the least chance of problems.

Reply:
I think by not returning the same instance of the context we don't respect one of the contract (maybe was just in my mind) that if the propagator fails to extract will return the same instance of the context so we can build a multi propagator that calls individual propagators until one succeeds

Reply:
It looks like our documentation does not suggest that, will check the specs as well

Reply:
Yes, that contract comes from the specs which we don't follow https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#extract

Reply:
I will think a bit about this, but it may be fine if we document just that if the passed context implementation may return `Root`, so we can still implement the logic I said.

But we definitely need to fix our extract contract.

Reply:
I think by returning `root` we are currently satisfying that spec

`If a value can not be parsed from the carrier, for a cross-cutting concern, the implementation MUST NOT throw an exception and MUST NOT store a new value in the Context, in order to preserve any previously existing valid value.`

We don't throw an exception nor store a new value in the Context since we're returning root. Semantically we can just consider null context and root context the same.

Is the concern addressed by adding `or empty context` in the behavior describing the unparsable here?

https://github.com/open-telemetry/opentelemetry-java/blob/main/api/context/src/main/java/io/opentelemetry/context/propagation/TextMapPropagator.java#L114